ourjapanlife/findadoc-server
# 27 postgres [draft]

Part I of #27 and relates to Dockerization: #28 

- [ ] Add Migration tool
- [x] Add ORM (object relational mapper): I chose Prisma because it works well with TS and has active develpment
- [x] Add postgres
- [x] Basic docker setup for postgres
- [x] Add .env secrets and set up GitHub repository secrets for this repo
- [ ] wire everything together

I'm trying to map the 1:1, 1:M, M:M relationships and have a rough draft here: https://miro.com/app/board/uXjVP3xzRlk=/?share_link_id=202260966339 

## Testing instructions

- Check the README under Database Setup
- Follow the instructions regarding Docker installation and Docker compose commands
- Right now, not everything is working... I will update these steps as I improve the draft
- TODO: test running migrations and connecting to the DB
